### DISCUSSION TITLE
Is there a way to move the form action buttons below the relation manager?

### DISCUSSION THREAD

**Discussion by maxacarvalho** [filamentphp/filament#6680] [Help]
Hi,

I have a scenario where users **must** update items (from the relation manager) before they save the form itself (owner record).  
It would feel more natural to have the form action buttons to be presented after the relation manager (the page will only have one relation manager, always).

I tried to use a custom view for the edit page, but when I moved the form action buttons below the relation manager, they stopped working. The save button is dead, and won't submit the form.

Any ideas?

Thank you.

**Comment by pboivin** [comment]
Hi @maxacarvalho, it makes sense that the button doesn't work because it's now outside of the form element it's trying to submit. Not sure if there's a clean way to accomplish what you want but I can think of a hack. In your custom view:

- Add an id to the form (e.g. `id="edit-record-form"`)

- Visually hide the `x-filament::form.actions` component (e.g. `class="hidden"`)

- Before the closing `</x-filament::page>` tag, add a new button that can trigger the `save` action inside the form :

```html
<x-filament-support::button
    x-on:click="document.querySelector('#edit-record-form button[type=submit]').click()"
>
    Save
</x-filament-support::button>
```

**Reply by maxacarvalho** [reply]
Hi @pboivin, thanks for taking time to reply.

I'd agree with what you said about the button being placed outside the form, except, it isn't (as far as I see).

This is the modified version of the edit page that I'm playing with:
```php 
<x-filament::page
    :widget-data="['record' => $record]"
    :class="\Illuminate\Support\Arr::toCssClasses([
        'filament-resources-edit-record-page',
        'filament-resources-' . str_replace('/', '-', $this->getResource()::getSlug()),
        'filament-resources-record-' . $record->getKey(),
    ])"
>
    @php
        $relationManagers = $this->getRelationManagers();
    @endphp

    <x-filament::form wire:submit.prevent="save">
        {{ $this->form }}

        <x-filament::resources.relation-managers
            :active-manager="$activeRelationManager"
            :form-tab-label="$this->getFormTabLabel()"
            :managers="$relationManagers"
            :owner-record="$record"
            :page-class="static::class"
        />

        <x-filament::form.actions
            :actions="$this->getCachedFormActions()"
            :full-width="$this->hasFullWidthFormActions()"
        />
    </x-filament::form>
</x-filament::page>
```

You can see that the `<x-filament::form.actions...` "lives" within the `<x-filament::form>` tag. So I'd expect that it would be considered to be *inside* the form.  
A fun fact is that if I place the `<x-filament::form.actions...` snippet **before** the `<x-filament::resources.relation-managers...` declaration, the submit button works as expected.

So it is the `<x-filament::resources.relation-managers...` piece of code that's disrupting the submit button. When the button snippet is placed below the relation manager.

**Reply by pboivin** [reply]
I see, this makes sense. I was assuming you pulled the `x-filament::form.actions` out of the form instead.

In your case, I'm not 100% sure but I think the `x-filament::resources.relation-managers` cannot be placed in the form because it contains a `<form>` tag itself, and nested form tags are not valid in HTML. Something weird must happen as a result of this and break the submit button.

**Reply by danharrin** [reply]
Also, I question why you would want the save button after relation managers. The relation managers are not saved at the same time as the rest of the form, so it would be a confusing UX. Also, it would bury the buttons and people could forget about them.

**Reply by maxacarvalho** [reply]
Hi @danharrin 

My scenario requires the user to:
1. Fill in the owner's data
2. Fill in missing data for the related items
3. Submit the form so a process that depends on the data (owner + children) starts.
